pkg/pxc/app/statefulset/node.go

```diff
@@ -404,91 +410,85 @@ func (c *Node) PMMContainer(ctx context.Context, cl client.Client, spec *api.PMM
 
 	ct.Env = append(ct.Env, pmmEnvs...)
 
-	if cr.CompareVersionWith("1.2.0") >= 0 {
-		clusterEnvs := []corev1.EnvVar{
-			{
-				Name:  "DB_CLUSTER",
-				Value: app.Name,
-			},
-			{
-				Name:  "DB_HOST",
-				Value: "localhost",
-			},
-			{
-				Name:  "DB_PORT",
-				Value: "3306",
-			},
-		}
-		ct.Env = append(ct.Env, clusterEnvs...)
-		ct.Resources = spec.Resources
-	}
-	if cr.CompareVersionWith("1.7.0") >= 0 {
-		for k, v := range ct.Env {
-			if v.Name == "DB_PORT" {
-				ct.Env[k].Value = "33062"
-				break
-			}
-		}
-		PmmPxcParams := ""
-		if spec.PxcParams != "" {
-			PmmPxcParams = spec.PxcParams
-		}
-		clusterPmmEnvs := []corev1.EnvVar{
-			{
-				Name:  "CLUSTER_NAME",
-				Value: cr.Name,
-			},
-			{
-				Name:  "PMM_ADMIN_CUSTOM_PARAMS",
-				Value: PmmPxcParams,
-			},
-		}
-		ct.Env = append(ct.Env, clusterPmmEnvs...)
-		pmmAgentScriptEnv := app.PMMAgentScript(cr, "mysql")
-		ct.Env = append(ct.Env, pmmAgentScriptEnv...)
+	clusterEnvs := []corev1.EnvVar{
+		{
+			Name:  "DB_CLUSTER",
+			Value: app.Name,
+		},
+		{
+			Name:  "DB_HOST",
+			Value: "localhost",
+		},
+		{
+			Name:  "DB_PORT",
+			Value: "3306",
+		},
 	}
-	if cr.CompareVersionWith("1.9.0") >= 0 {
-		fvar := true
-		ct.EnvFrom = []corev1.EnvFromSource{
-			{
-				SecretRef: &corev1.SecretEnvSource{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: cr.Spec.PXC.EnvVarsSecretName,
-					},
-					Optional: &fvar,
-				},
-			},
+	ct.Env = append(ct.Env, clusterEnvs...)
+	ct.Resources = spec.Resources
+
+	for k, v := range ct.Env {
+		if v.Name == "DB_PORT" {
+			ct.Env[k].Value = "33062"
+			break
 		}
+	}
+	PmmPxcParams := ""
+	if spec.PxcParams != "" {
+		PmmPxcParams = spec.PxcParams
+	}
```

Comment:
Given that `spec.PxcParams` will get a default value of an empty string during the spec initialization if it is not set, we can use it straightaway without this check, can you confirm?